test/e2e/mock-response-data/snaps/snap-binary-mocks.ts

```diff
@@ -0,0 +1,328 @@
+import fs from 'fs';
+import { escapeRegExp } from 'lodash';
+import { Mockttp } from 'mockttp';
+
+const SNAP_ASSETS_RELATIVE_PATH =
+  'test/e2e/mock-response-data/snaps/snap-binaries-and-headers';
+const NPM_REGISTRY_METAMASK_BASE_URL = 'https://registry.npmjs.org/@metamask';
+
+/**
+ * Sorts an array of file names by version strings semantically (e.g., 1.2.0, 1.10.0, 2.0.0).
+ * Modifies the array in place.
+ *
+ * @param versions - An array of version strings to sort.
+ */
+function sortFilesByVersion(versions: string[]): void {
+  versions.sort((a, b) => {
+    const partsA = a.split('.').map(Number);
+    const partsB = b.split('.').map(Number);
+    for (let i = 0; i < Math.max(partsA.length, partsB.length); i++) {
+      const valA = partsA[i] ?? 0;
+      const valB = partsB[i] ?? 0;
+      if (valA < valB) {
+        return -1;
+      }
+      if (valA > valB) {
+        return 1;
+      }
+    }
+    return 0;
+  });
```

Comment:
You can simplify this logic using [`semver`](https://www.npmjs.com/package/semver)'s `compare` function. That will also handle prereleases and other special cases properly.

Reply:
oh great point!! I didn't want to add any new dependency but I see we already have this in the project. Adding this change now :pray: